internal/users/manager.go

```diff
@@ -208,6 +271,14 @@ func (m *Manager) UpdateBrokerForUser(username, brokerID string) error {
 // UserByName returns the user information for the given user name.
 func (m *Manager) UserByName(username string) (UserEntry, error) {
 	usr, err := m.cache.UserByName(username)
+	if errors.Is(err, cache.NoDataFoundError{}) {
+		// Check if the user is a temporary user.
+		for uid, user := range m.temporaryUsers {
```

Comment:
Shouldn't we lock the mutex to access this?

Reply:
I don't think so. The purpose of the mutex is to avoid race conditions where the `temporaryUsers` are checked and then written based on the result, as it's done in `registerUser`. Here we only read the `temporaryUsers`, I don't see any race condition issues with that.

Reply:
done